server/medical-contact/util.ts

```diff
@@ -0,0 +1,40 @@
+import type {HydratedDocument} from 'mongoose';
+import type {MedicalContact, PopulatedMedicalContact} from './model';
+
+type MedicalContactResponse = {
+  _id: string;
+  owner: string;
+  title: string;
+  first_name: string;
+  last_name: string;
+  hospital: string;
+  specialty: string;
+  phone_number: string;
+  notes: string;
+};
+
+/**
+ * Transform a raw Medical Contact object from the database into an object
+ * with all the information needed by the frontend
+ *
+ * @param {HydratedDocument<MedicalContact>} medicalContact - A medical contact
+ * @returns {MedicalContactResponse} - The medical contact object formatted for the frontend
+ */
+const constructMedicalContactResponse = (medicalContact: HydratedDocument<MedicalContact>): MedicalContactResponse => {
+  const medicalContactCopy: PopulatedMedicalContact = {
+    ...medicalContact.toObject({
+      versionKey: false // Cosmetics; prevents returning of __v property
+    })
+  };
+  const {username} = medicalContactCopy.ownerId;
```

Comment:
related to the comment in the collections -- I don't think this is necessary since the user will always only see their own contacts

Reply:
I think just for data safety purposes, it makes sense either way to remove any password information from the frontend object?

Reply:
Sorry what I was meaning by this is to just delete the owner entirely in the response. No need to make an extra mongo call to populate the username when we already know what the username will be.

Reply:
in other words just delete lines 29 and 34

Reply:
since you already got rid of the `populate` calls I believe line 29 will crash as written currently

Reply:
oh I see, that makes sense, thank you!